src/notification.ts

```diff
@@ -89,47 +89,20 @@ export class BaseNotification {
     this.gateway = gateway;
 
     // Load fetch:
-    this.fetch = crossFetch;
-    this.setSessionFetch();
-
-    this.fetchLoaded = false;
-    this.fetchLoader = new Promise<void>((resolve) => {
-      if (fetchFn) {
-        this.setSessionFetch(fetchFn);
-        resolve();
-      } else {
-        // Attempt to load the fetch function from the default session if no fetchFn was passed in.
-        BaseNotification.getDefaultSessionFetch()
-          .then((defaultFetchFn) => {
-            if (defaultFetchFn) {
-              this.setSessionFetch(defaultFetchFn);
-            }
-          })
-          .finally(() => {
-            resolve();
-          });
-      }
-    }).then(() => {
-      this.fetchLoaded = true;
-    });
+    this.fetch = fetchFn || crossFetch;
+    if (!fetchFn) {
+      this.fetchLoaded = false;
+      this.fetchLoader = BaseNotification.getDefaultSessionFetch()
+        .then((defaultFetchFn) => {
+          if (defaultFetchFn) this.fetch = defaultFetchFn;
+        })
+        .catch(() => {})
```

Comment:
Is there something that should actually happen here? Could we sub in cross-fetch or something?